python_modules/dagster/dagster/_core/execution/stats.py

```diff
@@ -1,3 +1,5 @@
+# pyright: strict reportUnnecessaryIsInstance=false
```

Comment:
what's going on here

Reply:
`reportUnnecessaryIsIstance` is a pyright rule that is part of the `strict` ruleset. It flags when you perform an `isinstance` check that doesn't actually do any narrowing, because the type is already known by pyright.

To remove those unnecessary `isinstance` errors here requires runtime changes, so it's not done in this PR, but the file otherwise passes strict mode-- IMO this approach is more informative about the status of the file than leaving in type-ignores with or without a `# pyright: strict`.